service/topology_coordinator.cc

```diff
@@ -2008,6 +2016,11 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
     // Returns true if the state machine was transitioned into tablet migration path.
     future<bool> maybe_start_tablet_migration(group0_guard);
 
+    // Returns true if it successfully moved the upgrade forward.
```

Comment:
Good point. In some previous version of the code I used to start topology coordinator before upgrade could start, so in case of `not_started` I returned `false` - but now this should not happen and there is an `on_internal_error. I'll remove the bool from the return type.